### PR TITLE
feat: add eslint and prettier config

### PR DESCRIPTION
**Pull request by chrstph-dvx**
### Summary

Add eslint and prettier config that we can use in our various typescript repo.

### CODE REVIEW

File: packages/eslint-typescript/prettier.config.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
```
Comment (chrstph-dvx) [inline packages/eslint-typescript/prettier.config.js:1]: Once we publish the prettier config, we can extend from `"@arbitrum/prettier-config"` directly

File: README.md [hunk 2]
```diff
@@ -1 +1,54 @@
...
\ No newline at end of file
+# Eslint
+
+Provide reusable typescript and react eslint configs
+
+1. Install the peerDependencies of eslint configs:
+
+```sh
+npx install-peerdeps --dev @arbitrum/eslint-config-airbnb/typescript
+npx install-peerdeps --dev @arbitrum/eslint-config-airbnb/react
```
Comment (fionnachan) [inline README.md:9]: please update the dep name

Comment (fionnachan) [inline README.md]: please update the dep name 😬

**Review by fionnachan — COMMENTED** [review]
overall looks good! just needed one change on the README

File: packages/prettier/index.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  tabWidth: 2,
+  useTabs: false,
+  semi: true,
+  singleQuote: true,
+  bracketSpacing: true,
+  arrowParens: "avoid",
```
Comment (spsjvc) [inline packages/prettier/index.js]: Think I'd go with `always` here, as when you're changing an arrow function from one param to two, you have to first wrap with parentheses and then add the param (as opposed to just adding the param), which is kinda sucky

File: packages/prettier/index.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  trailingComma: "all",
```
Comment (spsjvc) [inline packages/prettier/index.js]: YESSSSSSSS

File: README.md [hunk 5]
```diff
@@ -1 +1,54 @@
...
+npx install-peerdeps --dev @arbitrum/eslint-config-typescript/react
+```
+
+2. Usage
+
+```ts
+module.exports = {
+  extends: [
+    "@arbitrum/eslint-config-typescript/typescript",
+    "@arbitrum/eslint-config-typescript/react",
```
Comment (spsjvc) [inline README.md]: A couple of nitpickings/questions: Should we maybe rename the package to `@arbitrum/typescript-eslint-config`, to better match `@typescript-eslint` and is it possible to do:

```suggestion
    "@arbitrum/typescript-eslint-config",
    "@arbitrum/typescript-eslint-config/react",
```

i.e. not having to do `/typescript` for the base package?

Comment (spsjvc) [reply README.md]: If not, maybe just call it `/base`? 🤷

**Review by spsjvc — COMMENTED** [review]
Love it!!

File: README.md [hunk 6]
```diff
@@ -1 +1,54 @@
-# config-monorepo
\ No newline at end of file
+# Eslint
+
+Provide reusable typescript and react eslint configs
+
+1. Install the peerDependencies of eslint configs:
+
+```sh
+npx install-peerdeps --dev @arbitrum/eslint-config-typescript/typescript
```
Comment (dewanshparashar) [inline README.md]: Is it possible that all `npx / npm `commands be replaced with yarn since we use that globally?

File: README.md (hunk 2 above)
Comment (chrstph-dvx) [reply README.md]: updated

File: README.md (hunk 5 above)
Comment (chrstph-dvx) [reply README.md]: having `eslint-config` at the start is standard, so I would keep it that way. `/base` make sense yes

File: README.md (hunk 6 above)
Comment (chrstph-dvx) [reply README.md]: `npx` will detect that `yarn` is installed, and use it for the install. AFAIK there is no equivalent for yarn

Comment (dewanshparashar) [reply README.md]: Got it! Thanks for this info!

File: packages/eslint-typescript/prettier.config.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+  tabWidth: 2,
+  useTabs: false,
+  semi: true,
+  singleQuote: true,
+  bracketSpacing: true,
+  arrowParens: 'avoid',
```
Comment (spsjvc) [inline packages/eslint-typescript/prettier.config.js]: This is not in sync with the other Prettier config?

**Review by spsjvc — DISMISSED** [review]
Amazing, LGTM

File: packages/prettier/index.js [hunk 8]
```diff
@@ -4,6 +4,6 @@ module.exports = {
   semi: true,
   singleQuote: true,
   bracketSpacing: true,
-  arrowParens: "avoid",
-  trailingComma: "all",
-};
+  arrowParens: 'always',
+  trailingComma: 'all',
+}
```
Comment (fionnachan) [inline packages/prettier/index.js]: is this missing `;` because of the `semi: false` on `prettier.config.js` previously?

Comment (chrstph-dvx) [reply packages/prettier/index.js]: Updated, config is now in `prettier.config.js` and we reexport it in `index.js`

**Review by spsjvc — APPROVED** [review]

**Review by fionnachan — APPROVED** [review]

**Pull request merged**